src/globus_sdk/services/auth/identity_map.py

```diff
@@ -219,6 +225,15 @@ def __getitem__(self, key: str) -> Any:
         """
         if key not in self._cache:
             self._fetch_batch_including(key)
+        # if the key was in the cache, double-check that it's not in the unresolved
+        # usernames or IDs for the IdentityMap
+        # otherwise, when a cache is being shared between IdentityMap instances, it's
+        # possible to call out to Auth unnecessarily
+        else:
+            if key in self.unresolved_ids:
+                self.unresolved_ids.remove(key)
+            elif key in self.unresolved_usernames:
+                self.unresolved_usernames.remove(key)
```

Comment:
I want to circle back and check that this is the right place to do this bookkeeping. It seems like we should be checking for _any_ values being in the cache already when we are doing a lookup against Auth; not just on `__getitem__`.